framework/src/controller/application.js

```diff
@@ -114,7 +115,10 @@ class Application {
 	 * @throws Framework.errors.SchemaValidationError
 	 */
 	constructor(genesisBlock, config = {}) {
-		validator.validate(genesisBlockSchema, genesisBlock);
+		liskValidator.validate(genesisBlockSchema, genesisBlock);
+		if (liskValidator.errors) {
+			throw liskValidator.errors;
+		}
```

Comment:
Validation must always throw error in case of failure, its so much redundant to use `if` everywhere. If one don't want to throw error, then he can use `try/catch` for his code block.

Reply:
I think this is outside the scope of this PR for which only methods that were missing were added to this library. Please open an issue with suggestions for lisk-validator

Reply:
This feedback is related to the code introduced in this PR, so not out of scope. If you are replacing the previous implementation with new then this is important instead of adding a lot of conditions in this PR.

Reply:
As discussed I changed lisk-validator to wrap ajv for it to return any errors during validation and return them. Resolving and starting fixing/rewriting tests.